### PR TITLE
Added Selection Algorithm Using Quick sort in C, C++, README.md

### LINKED ISSUE #1245: Adding Selection algorithms in c,c++ and README.md
Select kth Smallest Element Using QuickSort (nlogk)

### PR DESCRIPTION
**Pull request by Anshuvats**
Fixes #1245 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Implementation of Select kth Smallest Element Using QuickSort in C, C++, README files.

#### Languages Used:
- C
- C++

#### Files Added:
-  Selection_Algorithm_Repo/README.md 
-  Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.c 
-  Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp

### CODE REVIEW

File: Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// O(n square ) in the worst case
+
+int partition(int * a, int low, int high);
+int selectKth(int * a, int low, int high, int k) {
+
+  if (low == high)
+    return a[low];
+  else {
+
+    // Call the partition to place the first pivot.
```
Comment (MastersAbh) [inline Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.c]: Please fix indentation of this file. It is a 2 space indentation. We are following 4 space one.

File: Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+// This uses partition function of QuickSort to find kth smallest element in an unsorted array in nlogk time (avg case)
+// O(n square ) in the worst case
+int partition(int *a,int low,int high);
+int selectKth(int *a,int low,int high,int k){
+
+    if(low==high)
+        return a[low];
+    else{
+         // Call the partition to place the first pivot.
+            int p  = partition(a,low,high);
```
Comment (MastersAbh) [inline Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp]: space after comma

File: Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            if(p==k){
```
Comment (MastersAbh) [inline Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp]: space after if

File: Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                return a[p];
+            }
+            else if(k<p){
+                return selectKth(a,low,p-1,k);
+            }
+            else{
```
Comment (MastersAbh) [inline Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp]: space after else

File: Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                return selectKth(a,p+1,high,k);
```
Comment (MastersAbh) [inline Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp]: space around + and ,

File: Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+int main(){
+
+int a[] = {9,5,1,3,4,-2};
+int n=sizeof(a)/sizeof(int);
+int i;
+
+// This should give Sorted output . Selecting elements one by one .
+for(i=0;i<6;i++){
```
Comment (MastersAbh) [inline Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp]: space around operators and after for

File: Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+int ans =  selectKth(a,0,n-1,i);
+cout<<ans<<endl;
```
Comment (MastersAbh) [inline Selection_Algorithm_Repo/Selection_algo_using_Quick_sort.cpp]: please follow indentation

**Comment by Anshuvats** [comment]
Hey @MastersAbh, I have followed up with the changes Request. Can You please check?

**Comment by rishabhgarg25699** [comment]
@Anshuvats Please try to squash your commit .

**Comment by Anshuvats** [comment]
Hey @rishabhgarg25699, I have squashed the commit. Can you please check?

**Pull request merged**